### ISSUE TITLE
Copy Application Package to Image Store hangs forever.

### ISSUE THREAD

**Issue by SteppingRazor** [Azure/service-fabric-issues#1170]
Hello,
I have a standalone cluster on 3 Windows Server 2016 machines. When I'm trying to deploy the application package to the cluster it hangs forever on step:
`Copying application to image store...`
And this happens almost all the time. I said almost because when I spam deployment script few times, one of them finally get pass copy step. Also during the copy I can see these warnings on my machine event-viewer from where I do the deployment:

`0923a8e5-75f9-4f05-91a6-e806f36020e6: SendOneWay during UploadBytes failed with TransportSendQueueFull. Retrying after 3000 milliseconds.`

`client-dev.net:19000: 21 (66070125 bytes) in queue, limit=67108864B, dropping messge {00000000-0000-0000-0000-000000000000}:0(3146087B), Actor = FileReceiver, Action = 'FileContent'`

`71468d66-2861-462e-888e-563c0d747226: SendOneWay during UploadBytes failed with TransportSendQueueFull. Retrying after 3000 milliseconds.`

Any ideas what can be wrong? How can I investigate it? Any additional logs I can take a look? Especially the fact that sometimes it works after few tries is super suspicious.

**Comment by masnider** [comment]
I've seen things like this before when there were firewall issues blocking communication between the nodes in the cluster. Did the [BPA](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-cluster-creation-for-windows-server) say the setup was all correct?

**Comment by SteppingRazor** [comment]
Even if it would be a firewall does it explain that sometimes it works? 
BPA was ok.

**Comment by oanapl** [comment]
@SteppingRazor , apologies for the delay. Is this still an issue?

What Service Fabric version do you have on the client machine and in the cluster (assuming the client runs outside the cluster)?

What are you using for deployment (Powershell scripts, VSTS etc)? Have you tried with multiple application packages and all fail this way (get stuck)? What timeout do you pass to copy application? How big & how many files are in your package?

Is fabric:/System/ImageStoreService healthy in the cluster? Is [SMB properly configured](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-cluster-standalone-deployment-preparation#environment-setup)? 

Did you notice any Warning/Errors in the event viewer that correlate with your package upload (related to SMB, FileStoreService etc)?

**Comment by SteppingRazor** [comment]
@oanapl 

> Is this still an issue?

Yes

> What Service Fabric version do you have on the client machine and in the cluster (assuming the client runs outside the cluster)?

Cluster nodes machines are running the same version as "installation machine" from which we are trying to deploy the package. 
SF Runtime Version on all machines: 6.2.283.9494
SDK version installed on external installation machine: 3.1.283.9494

> What are you using for deployment (Powershell scripts, VSTS etc)?

We are using default _Deploy-FabricApplication.ps1_ with some small adjustments. 
But using "pure" Service Fabric _Copy-ServiceFabricApplicationPackage_ command is causing the same error as well.

> Have you tried with multiple application packages and all fail this way (get stuck)?

Yes

> What timeout do you pass to copy application?

Tried maximum allowed number, didn't help.

> How big & how many files are in your package?

Compressed 187MB - 55 files

> Is fabric:/System/ImageStoreService healthy in the cluster?

Yes

> Is SMB properly configured?

Yes

> Did you notice any Warning/Errors in the event viewer that correlate with your package upload (related to SMB, FileStoreService etc)?

Besides the ones mentioned on my first post, no.

**Comment by oanapl** [comment]
At this point, the fastest way to make progress is to [create a support request](https://ms.portal.azure.com/#blade/Microsoft_Azure_Support/HelpAndSupportBlade/overview).

Another option is to collect the Service Fabric logs from all machines involved (cluster + client) and attach them here or send them to me (oanapl at microsoft.com). (assuming Windows) The reg key HKLM\SOFTWARE\Microsoft\ServiceFabric\FabricLogRoot should point to the location of the logs.

**Comment by oanapl** [comment]
One thing you can check before collecting logs or creating support ticket, is to look at MaxMessageSize in your cluster and the client machine. See #813 for a similar issue.

**Comment by cfe84** [comment]
I have a similar issue when deploying to any remote SF cluster, with following events in the Microsoft Service Fabric / Admin event list:

```
client-xxxxxxx.eastus.cloudapp.azure.com:19000: 50 (66669561 bytes) in queue, limit=67108864B, dropping messge {d1cf42e2-4a86-4863-ad44-6833dd1e7f1f}:751(3145958B), Actor = FileReceiver, Action = 'FileContent'
```

```
5cad060e-de3c-4d46-a35b-9f2ff37fc52e: SendOneWay during UploadFileChunk failed with retryable error:TransportSendQueueFull retryCount:116 storeRelativePath:ComputerzType\Computerz.APIPkg\Code.zip.
```

Just did an update of SF SDK, wonder if it's linked.

**Comment by oanapl** [comment]
@cfe84 , those are retryable errors and should not lead to failure in normal case. 

@SteppingRazor opened a support ticket and our CSS is helping him look at the issue. We saw some processes crash on one of the nodes, and the support engineer is trying to collect dumps and logs at the time of failure.

**Comment by iyyappam** [comment]
@cfe84 , What is the size of your application package? I see that you were using compress option, could you try without compress and check it out.

**Comment by WhitWaldo** [comment]
I'm experiencing this issue as well, but only within my VSTS instance. I can deploy from my local dev machine without an issue (or at least have successfully done so in the last two of two attempts today).

@iyyappam I can answer your question with regards to myself: the /drop/applicationpackage directory on my build server is 1 GB in size. I've attempted to deploy using both compress and without from VSTS with no noticeable difference in success rate.

To @oanapl 's suggestion earlier, I've set the MaxMessageSize option to 10 MB on the cluster and haven't seen that it had an impact on this either.

**Comment by oanapl** [comment]
@WhitWaldo , stuck copy can happen due to many reasons (eg. cluster not properly setup, other configuration issue, Service Fabric bug, VSTS bug etc). There are already 3 issues followed on this thread and it's getting hard to follow them. Please use [the thread you already opened with VSTS](https://github.com/Microsoft/vsts-tasks/issues/8145) to follow up; if you need a Service Fabric issue, please open a new one.

@cfe84 and @SteppingRazor , we haven't heard from you in a while. I will close this issue, feel free to re-open (or start new thread) if you are still having issues.
